File: editor/en/[2125]Number of Laser Beams in a Bank.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string[]} bank
 * @return {number}
 */
var numberOfBeams = function (bank) {
    let prev = 0;
    let total = 0;

    for (let row of bank) {
        let curr = getCount(row);

        if (curr === 0) {
            continue;
        }

        total += curr * prev;
        prev = curr;
    }

    return total;

    function getCount(s) {
        let count = 0;

        for (let c of s) {
            count += parseInt(c);
        }

        return count;
    }
};
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2125]Number of Laser Beams in a Bank.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int numberOfBeams(String[] bank) {
        int prev = 0;
        int total = 0;

        for (String row : bank) {
            int curr = getCount(row);

            if (curr == 0) {
                continue;
            }

            total += curr * prev;
            prev = curr;
        }

        return total;
    }

    private int getCount(String s) {
        int count = 0;

        for (char c : s.toCharArray()) {
            count += c - '0';
        }

        return count;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
